docs/source/torch.compiler_deepdive.rst

```diff
@@ -1,7 +1,50 @@
-TorchDynamo Deeper Dive
-=======================
+TorchDynamo Deep Dive
+=====================
+
+Before you read this section, read :ref:`torch.compiler_overview`.`
+
+**TorchDynamo** is a Python-level Just-In-Time (JIT) compiler designed to make
+unmodified PyTorch programs faster. TorchDynamo hooks into the frame evaluation
+API in CPython (`PEP 523 <https://peps.python.org/pep-0523/>`__) to
+dynamically modify Python bytecode right before it is executed. It
+rewrites Python bytecode to extract sequences of PyTorch
+operations into an `FX Graph <https://pytorch.org/docs/stable/fx.html>`__
+which is then compiled with a customizable backend.
+It creates this FX Graph through bytecode analysis and is designed to
+mix Python execution with compiled backends to get the best of both
+worlds — usability and performance.
+
+TorchDynamo makes it easy to experiment with different compiler
+backends to make PyTorch code faster with a single line decorator
+``torch._dynamo.optimize()`` which is wrapped for convenience by ``torch.compile()``
+
+The following diagram demonstrates how PyTorch works with ``torch.compile``
+and without it:
+
+.. image:: _static/img/dynamo/TorchDynamo.png
+
+`TorchInductor` is one of the backends
+supported by `TorchDynamo Graph <https://pytorch.org/docs/stable/fx.html>`__
+into `Triton <https://github.com/openai/triton>`__ for GPUs or
+`C++/OpenMP <https://www.openmp.org/>`__ for CPUs. We have a
+`training performance dashboard <https://github.com/pytorch/torchdynamo/issues/681#issuecomment-1233828468>`__
+that provides performance comparison for different training backends. You can read
+more in the `TorchInductor post on PyTorch
+dev-discuss <https://dev-discuss.pytorch.org/t/torchinductor-a-pytorch-native-compiler-with-define-by-run-ir-and-symbolic-shapes/747>`__.
+
+For an in-depth overview, read the sections below, watch the deep-dive video,
+and check out the dev-discuss topics.
+
+   * `TorchDynamo deep-dive video <https://www.youtube.com/watch?v=egZB5Uxki0I>`__
+   * `dev-discuss topics <https://dev-discuss.pytorch.org/search?q=TorchDynamo%20order%3Alatest>`__
+
+TorchDynamo Internals
+~~~~~~~~~~~~~~~~~~~~~
 **Author**: `Jason Ansel <https://github.com/jansel>`_
 
+This section will go over some of the TorchDynamo internals and will
```

Comment:
There was a user complaint that the guard section was missing what code snippet it was referring to, this was what the code looked like before then https://github.com/pytorch/torchdynamo/pull/1118/files

Reply:
I guees it's about this:
> For example, the printouts for the first graph in the `toy_example`
   above are:

And there is no toy_example above. I guess this is the right one: https://github.com/pytorch/pytorch/blob/main/docs/source/compile/guards-overview.rst#L24

Reply:
I guess the toy_example from here is missing. I'll copy it here.